[jit] Switch Value::setDebugName() from stringstream to fmt::format

jjlilley opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42820 [jit] Switch Value::setDebugName() from stringstream to fmt::format**

This format call is taking ~11% of a data_preproc/perf_test benchmark.

If I microbenchmark both the stringstream version and the libfmt version,
libfmt is better than 10x faster:
   (For 1000 iterations, so really 244ns vs 21ns)
   WithStringStream    244.38us    4.09K
   WithLibFmt           21.89us   45.67K

Differential Revision: [D23033537](https://our.internmc.facebook.com/intern/diff/D23033537/)

Commit:
[jit] Switch Value::setDebugName() from stringstream to fmt::format

This format call is taking ~11% of a data_preproc/perf_test benchmark.

If I microbenchmark both the stringstream version and the libfmt version,
libfmt is better than 10x faster:
   (For 1000 iterations, so really 244ns vs 21ns)
   WithStringStream    244.38us    4.09K
   WithLibFmt           21.89us   45.67K

Differential Revision: [D23033537](https://our.internmc.facebook.com/intern/diff/D23033537/)

[ghstack-poisoned]

jjlilley commented:
Hello, is there a better reviewer for this diff? (or an alternate approach propose?) Thanks!

suo reviewed (approved):
Thanks, was meaning to do this!

Commit:
rebase on "[jit] Switch Value::setDebugName() from stringstream to fmt::format"


This format call is taking ~11% of a data_preproc/perf_test benchmark.

If I microbenchmark both the stringstream version and the libfmt version,
libfmt is better than 10x faster:
   (For 1000 iterations, so really 244ns vs 21ns)
   WithStringStream    244.38us    4.09K
   WithLibFmt           21.89us   45.67K

Differential Revision: [D23033537](https://our.internmc.facebook.com/intern/diff/D23033537/)

[ghstack-poisoned]
